src/wallet/api.rs

```diff
@@ -174,17 +174,18 @@ impl Wallet {
             master_key,
             Some(wallet_birthday),
         )?;
+
         Ok(Self {
             rpc,
-            wallet_file_path: path.clone(),
+            wallet_file_path: path.to_path_buf(),
```

Comment:
cant we just say `path.as_ref().to_path_buf()`? Then remove the `path = path.as_ref()` line above.